Index: audio_device_utility_ios.cc
===================================================================
diff --git a/trunk/webrtc/modules/audio_device/ios/audio_device_utility_ios.cc b/trunk/webrtc/modules/audio_device/ios/audio_device_utility_ios.cc
--- a/trunk/webrtc/modules/audio_device/ios/audio_device_utility_ios.cc	(revision 7089)
+++ b/trunk/webrtc/modules/audio_device/ios/audio_device_utility_ios.cc	(working copy)
@@ -27,8 +27,9 @@
 AudioDeviceUtilityIPhone::~AudioDeviceUtilityIPhone() {
     WEBRTC_TRACE(kTraceMemory, kTraceAudioDevice, _id,
                  "%s destroyed", __FUNCTION__);
-    CriticalSectionScoped lock(&_critSect);
-
+    {
+        CriticalSectionScoped lock(&_critSect);
+    }
     delete &_critSect;
 }
 
