Index: iossim.gyp
===================================================================
diff --git a/trunk/testing/iossim/iossim.gyp b/trunk/testing/iossim/iossim.gyp
--- a/trunk/testing/iossim/iossim.gyp	(revision 288251)
+++ b/trunk/testing/iossim/iossim.gyp	(working copy)
@@ -25,50 +25,6 @@
               'defines': [
                 'IOSSIM_USE_XCODE_6',
               ],
-              'actions': [
-                {
-                  'action_name': 'generate_dvt_foundation_header',
-                  'inputs': [
-                    '<(iphone_sim_path)/DVTFoundation.framework/Versions/Current/DVTFoundation',
-                    '<(PRODUCT_DIR)/class-dump',
-                  ],
-                  'outputs': [
-                    '<(INTERMEDIATE_DIR)/iossim/DVTFoundation.h'
-                  ],
-                  'action': [
-                    # Actions don't provide a way to redirect stdout, so a custom
-                    # script is invoked that will execute the first argument and
-                    # write the output to the file specified as the second argument.
-                    # -I sorts classes, categories, and protocols by inheritance.
-                    # -C <regex> only displays classes matching regular expression.
-                    './redirect-stdout.sh',
-                    '<(PRODUCT_DIR)/class-dump -CDVTStackBacktrace|DVTInvalidation|DVTMixIn <(iphone_sim_path)/DVTFoundation.framework',
-                    '<(INTERMEDIATE_DIR)/iossim/DVTFoundation.h',
-                  ],
-                  'message': 'Generating DVTFoundation.h',
-                },
-                {
-                  'action_name': 'generate_dvt_core_simulator',
-                  'inputs': [
-                    '<(developer_dir)/Library/PrivateFrameworks/CoreSimulator.framework/Versions/Current/CoreSimulator',
-                    '<(PRODUCT_DIR)/class-dump',
-                  ],
-                  'outputs': [
-                    '<(INTERMEDIATE_DIR)/iossim/CoreSimulator.h'
-                  ],
-                  'action': [
-                    # Actions don't provide a way to redirect stdout, so a custom
-                    # script is invoked that will execute the first argument and
-                    # write the output to the file specified as the second argument.
-                    # -I sorts classes, categories, and protocols by inheritance.
-                    # -C <regex> only displays classes matching regular expression.
-                    './redirect-stdout.sh',
-                    '<(PRODUCT_DIR)/class-dump -CSim <(developer_dir)/Library/PrivateFrameworks/CoreSimulator.framework',
-                    '<(INTERMEDIATE_DIR)/iossim/CoreSimulator.h',
-                  ],
-                  'message': 'Generating CoreSimulator.h',
-                },
-              ],  # actions
             }, {  # else: xcode_version!="6"
               'variables': {
                 'iphone_sim_path': '<(developer_dir)/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks',
@@ -81,36 +37,9 @@
           'include_dirs': [
             '<(INTERMEDIATE_DIR)/iossim',
           ],
-          'sources': [
-            'iossim.mm',
-            '<(INTERMEDIATE_DIR)/iossim/iPhoneSimulatorRemoteClient.h',
-          ],
           'libraries': [
             '$(SDKROOT)/System/Library/Frameworks/Foundation.framework',
           ],
-          'actions': [
-            {
-              'action_name': 'generate_dvt_iphone_sim_header',
-              'inputs': [
-                '<(iphone_sim_path)/DVTiPhoneSimulatorRemoteClient.framework/Versions/Current/DVTiPhoneSimulatorRemoteClient',
-                '<(PRODUCT_DIR)/class-dump',
-              ],
-              'outputs': [
-                '<(INTERMEDIATE_DIR)/iossim/DVTiPhoneSimulatorRemoteClient.h'
-              ],
-              'action': [
-                # Actions don't provide a way to redirect stdout, so a custom
-                # script is invoked that will execute the first argument and
-                # write the output to the file specified as the second argument.
-                # -I sorts classes, categories, and protocols by inheritance.
-                # -C <regex> only displays classes matching regular expression.
-                './redirect-stdout.sh',
-                '<(PRODUCT_DIR)/class-dump -I -CiPhoneSimulator <(iphone_sim_path)/DVTiPhoneSimulatorRemoteClient.framework',
-                '<(INTERMEDIATE_DIR)/iossim/DVTiPhoneSimulatorRemoteClient.h',
-              ],
-              'message': 'Generating DVTiPhoneSimulatorRemoteClient.h',
-            },
-          ],  # actions
           'xcode_settings': {
             'ARCHS': ['x86_64'],
             'WARNING_CFLAGS': [
